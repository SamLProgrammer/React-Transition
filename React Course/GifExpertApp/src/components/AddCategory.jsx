import { useState } from "react"
import PropTypes from 'prop-types'
export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('One Punch');

    const handleInputChange = (evt) => {
        setInputValue((v) => evt.target.value);
    }

    const handleFormSubmit = (evt) => {
        evt.preventDefault();

        const value = inputValue.trim();
        if(value.length < 1) return;
        onAddCategory(value);
        setInputValue((vl) => {return ''});
    }

    return (
        <form aria-label="form" onSubmit={ (event) => {handleFormSubmit(event)} }>
            <input type="text"
                placeholder="Search Gif"
                onChange={handleInputChange}
                value={inputValue} />
        </form>
    )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
