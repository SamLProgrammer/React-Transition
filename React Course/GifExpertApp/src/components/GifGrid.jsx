import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types'

export const GifGrid = ({ category, limit }) => {

    const {gifs, isLoading} = useFetchGifs(category, limit);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<p>Loading...</p>)
            }
            <div className="card-grid">
                {
                    gifs.map(( img ) => <GifItem key={img.key} {...img} />)
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limit: PropTypes.number.isRequired,
}
