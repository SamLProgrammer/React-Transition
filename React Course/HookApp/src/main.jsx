import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { MainApp } from './useContext/MainApp'
import './useReducer/intro-reducer'
import { BrowserRouter, createBrowserRouter, RouterProvider, } from "react-router-dom";

ReactDOM.createRoot(document.getElementById('root')).render(
    //   // <React.StrictMode>
    <BrowserRouter>
        <MainApp />
    </BrowserRouter>
    //   //  </React.StrictMode>,
)
