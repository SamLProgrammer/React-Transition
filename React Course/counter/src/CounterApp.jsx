import PropTypes from 'prop-types'
import { useState } from 'react';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);
    
    const handleAddClick = (evt) => {
        setCounter((c) => c+1);
    }

    const handleSubClick = (evt) => {
        setCounter((c) => c-1);
    }

    const handleResetClick = (evt) => {
        setCounter(value);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>
            <button onClick={(evt) => handleAddClick(evt)}>+1</button>
            <button onClick={(evt) => handleSubClick(evt)}>-1</button>
            <button onClick={(evt) => handleResetClick(evt)}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}