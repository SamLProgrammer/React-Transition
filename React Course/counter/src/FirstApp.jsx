import PropTypes from 'prop-types'

export const FirstApp = ({ title, subTitle }) => {
    return (
        <>
            <h1>First App</h1>
            <h2>Hey {title}</h2>
            <h3>You are: {subTitle}</h3>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}