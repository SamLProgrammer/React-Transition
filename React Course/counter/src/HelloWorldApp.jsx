export const App = () => {
    return (
        <h1>Hola ke ase!!</h1>
    );
}

