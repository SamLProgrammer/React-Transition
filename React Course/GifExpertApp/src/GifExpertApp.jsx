import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon Ball']);

    const handleAddCategory = (category) => {
        if (categories.find((cat) => { return cat.toLowerCase() == category.toLowerCase() })) return;
        setCategories((cat) => { return [category, ...categories] });
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory onAddCategory={handleAddCategory} />
            {
                categories.map((category) => <GifGrid category={category} limit={5} key={category}/>)
            }
        </>
    )
}
