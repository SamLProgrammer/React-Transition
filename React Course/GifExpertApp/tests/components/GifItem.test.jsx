import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('GifItemTest', () => {
    test('Should Ask for Title and URl', () => {
        const title = 'Broly';
        const url = 'http://broly.com/broly.jpg';
        const {container} = render(<GifItem url={url} title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Should ahow image with given url and tilte', () => {
        const title = 'Broly';
        const url = 'http://broly.com/broly.jpg';
        render(<GifItem url={url} title={title} />);
        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);
    });

    test('Component should display title', () => { 
        const title = 'Broly';
        const url = 'http://broly.com/broly.jpg';
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
     })
})