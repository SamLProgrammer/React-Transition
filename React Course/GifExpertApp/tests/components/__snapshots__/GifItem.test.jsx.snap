// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItemTest Should Ask for Title and URl 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Broly"
      src="http://broly.com/broly.jpg"
    />
    <p>
      Broly
    </p>
  </div>
</div>
`;
